Mathlib/Analysis/NormedSpace/Star/Matrix.lean

```diff
@@ -70,3 +86,79 @@ theorem entrywise_sup_norm_bound_of_unitary {U : Matrix n n 𝕜} (hU : U ∈ Ma
 #align entrywise_sup_norm_bound_of_unitary entrywise_sup_norm_bound_of_unitary
 
 end EntrywiseSupNorm
+
+noncomputable section EuclideanSpaceOpNorm
+
+variable [Fintype n] [DecidableEq n] [IsROrC 𝕜]
+
+/-- The natural star algebra equivalence between matrices and continuous linear endomoporphisms
+of Euclidean space induced by the orthonormal basis `EuclideanSpace.basisFun`. -/
+def Matrix.toEuclideanClm :
+    Matrix n n 𝕜 ≃⋆ₐ[𝕜] (EuclideanSpace 𝕜 n →L[𝕜] EuclideanSpace 𝕜 n) :=
+  LinearMap.toMatrixOrthonormal (EuclideanSpace.basisFun n 𝕜) |>.symm.trans <|
+    { LinearMap.toContinuousLinearMap with
+      map_mul' := fun _ _ ↦ rfl
+      map_star' := LinearMap.adjoint_toContinuousLinearMap }
+
+/-- An auxiliary definition used only to construct the true `NormedRing` (and `Metric`) structure
+provided by `Matrix.instMetricSpaceCstar` and `Matrix.instNormedRingCstar`.  -/
+def Matrix.NormedRingAuxCstar : NormedRing (Matrix n n 𝕜) :=
+  @NormedRing.induced ((Matrix n n 𝕜) ≃⋆ₐ[𝕜] (EuclideanSpace 𝕜 n →L[𝕜] EuclideanSpace 𝕜 n))
+    _ _ _ ContinuousLinearMap.toNormedRing _ _ Matrix.toEuclideanClm.injective
+
+open Bornology Filter
+open scoped Topology Uniformity
+
+/-- The metric on `Matrix n n 𝕜` arising from the operator norm given by the identification with
+(continuous) linear endmorphisms of `EuclideanSpace 𝕜 n`. -/
+def Matrix.instMetricSpaceCstar : MetricSpace (Matrix n n 𝕜) := by
```

Comment:
Does this generalize to rectangular matrices too?

Reply:
yes, I have now rewritten to generalize everything to rectangular matrices that can be so generalized.

Reply:
It looks like you haven't pushed?

Reply:
Indeed, I meant to finish up, but got distracted.